ipatests/test_integration/test_epn.py

```diff
@@ -344,22 +349,114 @@ def test_EPN_smoketest_3(self):
 
-    def test_EPN_nbdays(self):
+    def test_EPN_nbdays_0(self, cleanupmail):
         """Test the to/from nbdays options (implies --dry-run)
 
            We have a set of users installed with varying expiration
            dates. Confirm that to/from nbdays finds them.
+
+           Make sure --dry-run does not accidentally send emails.
         """
 
-        # Compare the notify_ttls values
+        # Use the notify_ttls values with a 1-day sliding window
         for i in self.notify_ttls:
             user_list = []
-            (stdout_text_client, unused) = self._check_epn_output(
-                self.clients[0], from_nbdays=i, to_nbdays=i + 1, dry_run=True)
+            (stdout_text_client, unused, unused) = self._check_epn_output(
+                self.clients[0], from_nbdays=i, to_nbdays=i + 1, dry_run=True
+            )
             for user in json.loads(stdout_text_client):
                 user_list.append(user["uid"])
             assert len(user_list) == 1
-            assert user_list[0] == "user%d" % i
+            userid = "user{id}".format(id=i)
+            assert user_list[0] == userid
+
+            # Check that the user list is expected for any given notify_ttls.
+            (stdout_text_client, unused, unused) = self._check_epn_output(
+                self.clients[0], to_nbdays=i
+            )
+            user_list = [user["uid"] for user in json.loads(stdout_text_client)]
+            assert len(user_list) == 1
+            assert user_list[0] == "user{id}".format(id=i - 1)
+
+            # make sure no emails were sent
+            result = self.clients[0].run_command(['ls', '-lha', '/var/mail/'])
+            assert ".." in result.stdout_text
+            assert userid not in result.stdout_text
+
+    def test_EPN_nbdays_1(self, cleanupmail):
+        """Test that for a given range, we find the users in that range"""
+
+        # Use hardcoded date ranges for now
+        for date_range in [(0, 5), (7, 15), (1, 20)]:
+            expected_user_list = ["user{i}".format(i=i)
+                                  for i in range(date_range[0], date_range[1])]
+            (stdout_text_client, unused, unused) = self._check_epn_output(
+                self.clients[0],
+                from_nbdays=date_range[0],
+                to_nbdays=date_range[1]
+            )
+            user_list = [user["uid"] for user in json.loads(stdout_text_client)]
+            for user in expected_user_list:
+                assert user in user_list
+            for user in user_list:
+                assert user in expected_user_list
+
+    # Test the to/from nbdays options behavior with illegal input
+
+    def test_EPN_nbdays_input_0(self):
+        """Make sure that --to-nbdays implies --dry-run."""
+
+        (stdout_text_client, unused, unused) = self._check_epn_output(
+            self.clients[0], to_nbdays=5, dry_run=False
+        )
+        assert len(json.loads(stdout_text_client)) >= 1
```

Comment:
Do we care how many lines were returned, what is in those lines or whether the output is valid json?

Reply:
That the output is valid json and contains at least one entry: essentially "Make sure that --to-nbdays implies --dry-run."

Reply:
I think the test would be better if it verified that the expected output matches what is seen.

Reply:
That's more or less what test_EPN_nbdays_1 does. Do you want it in test_EPN_nbdays_input_0 too?

Reply:
I think think that depending on the length of the output and not the content of the output is fragile.

Reply:
OK, agreed, will add the required logic there too.

Reply:
Done.